gui/src/main/scala/co/uproot/abandon/RegReport.scala

```diff
@@ -1,63 +1,110 @@
 package co.uproot.abandon
 
+import scala.util.matching.Regex
 import scalafx.Includes._
-import scalafx.scene.control.TreeView
-import scalafx.scene.control.TreeItem
-import scalafx.scene.input.KeyEvent
-import scalafx.scene.control.TreeCell
-import scalafx.stage.Stage
+import scalafx.application.Platform
 import scalafx.scene.Scene
-import scalafx.scene.control.ScrollPane
-import scalafx.stage.Modality
+import scalafx.scene.control.{ScrollPane, TreeCell, TreeItem, TreeView}
+import scalafx.scene.input.KeyEvent
+import scalafx.stage.{Modality, Stage}
 
 object RegUIReport extends UIReport {
-  private def getNestedTxns(item: TreeItem[RegisterReportEntry]): Seq[DetailedPost] = {
-    item.getValue.txns ++ item.children.flatMap(getNestedTxns(_))
+
+  case class TxnItem(txnGroup: String, reportEntry: RegisterReportEntry)
+
+  implicit class TxnItemMatcher(t: TxnItem) {
+    val accountMatcher: Regex = """[^\s]+""".r
+    def matches(item: TreeItem[TxnItem]): Boolean = {
+      isMatchingParent(item) || isMatchingTxnGroup(item)
+    }
+    private def isMatchingParent(item: TreeItem[TxnItem]) = {
+      t.reportEntry.txns.isEmpty && t.reportEntry.render == item.getValue.reportEntry.render
+    }
+    private def isMatchingTxnGroup(item: TreeItem[TxnItem]) = {
+      t.reportEntry.txns.nonEmpty && t.txnGroup == item.getValue.txnGroup &&
+        accountName(t.reportEntry) == accountName(item.getValue.reportEntry)
+    }
+    private def accountName(entry: RegisterReportEntry): String = {
+      accountMatcher.findFirstIn(t.reportEntry.render).getOrElse(t.reportEntry.render)
+    }
   }
 
+  private def getNestedTxns(item: TreeItem[TxnItem]): Seq[DetailedPost] = {
+    filterSelectedDates(item.getValue.reportEntry.txns) ++ item.children.flatMap(getNestedTxns(_))
+  }
+  def filterSelectedDates(txns: Seq[DetailedPost]): Seq[DetailedPost] = {
+    if (selectedTxnDates.isEmpty) txns
+    else txns.filter(txn => selectedTxnDates.contains(txn.date))
```

Comment:
This will only show newer transactions if they are on the exact same date as one of the previous transactions, right? That is not how I expect it to work. The transactions view is supposed to show all transactions for the selected date range and account name.

I am sorry if the expectation was not clear; the documentation is poor in this project.